x-pack/plugins/task_manager/task_manager.ts

```diff
@@ -85,10 +86,25 @@ export class TaskManager {
     this.poller = poller;
 
     kbnServer.afterPluginsInit(async () => {
-      this.isInitialized = true;
       store.addSupportedTypes(Object.keys(this.definitions));
-      await store.init();
-      await poller.start();
+      const startPoller = () => {
+        return new Promise(resolve => {
```

Comment:
Given that `afterPluginsInit` is `async`, if you just returned this instead of making the function, then calling it, then this would honor the code contract and allow you to specify `resolve(true)` as `this.isInitialized = true`.

Unless you were worried about this holding up other plugin initialization (it kind of should?)?

Reply:
Taking another look, you still need the function (for the `setTimeout` call), but you can still simplify it a little and return it (effectively `await`) if you wanted.

Reply:
5ece72fd1a